CMakeLists.txt

```diff
@@ -1379,13 +1379,22 @@ list_append_ifdef(
   COMMAND ${CMAKE_COMMAND} -E copy ${KERNEL_ELF_NAME}    ${KERNEL_EXE_NAME}
   )
 
+list_append_ifdef(
+  CONFIG_CUSTOM_IMAGE_BINARY
+  post_build_commands
+  COMMAND ${PYTHON_EXECUTABLE} ${BOARD_DIR}/support/create_board_img.py
+    -i ${PROJECT_BINARY_DIR}/zephyr.bin
+    -o ${PROJECT_BINARY_DIR}/zephyr_${BOARD}.bin
```

Comment:
I am concerned that we will be polluting core build scripts, like zephyr/CMakeLists.txt with abstractions that are not flexible enough to be re-used many platforms.

e.g., what if a board needs to modify a hex file?
Which binary will a user expect to be flashed when he invokes "ninja flash" ?

Would you consider instead introducing this patch to zephyr/CMakeLists.txt:

```
get_property(
  extra_post_build_commands
  GLOBAL PROPERTY
  extra_post_build_commands
  )

list(APPEND
  post_build_commands
  ${extra_post_build_commands}
  )

```

Then you can inject the post_build_command from a board build script without needing to create a flexible abstraction.

Reply:
Thanks for this suggestion, Sebastian. This is exactly what i wanted but didn't know how to do it. Will upload a revised patch.